glom/core.py

```diff
@@ -100,7 +107,42 @@ class GlomError(Exception):
     (e.g., ``len``, ``sum``, any ``lambda``) will not be wrapped in a
     GlomError.
     """
-    pass
+    @classmethod
+    def wrap(cls, exc):
+        class GlomWrapError(type(exc), GlomError): pass
+        return GlomWrapError(*exc.args)
+        # this is another pattern, but doesn't work for C-defined exceptions
+        # (maybe if hasattr(sys.modules[type(exc).__module__], "__file__")?)
+        exc2 = copy.copy(exc)
+        exc2.__class__ = GlomWrapError
+        return exc2
+
+    def _finalize(self, scope):
+        from . import trace
+
+        # tried MANY approaches here -- one problem pragmatically is
+        # that traceback module and sys.exc_info() see different
+        # stacks here -- maybe it is something to do with running
+        # under pytest
+        lines = traceback.format_exc().split("\n")
+        limit = 0
+        for line in reversed(lines):
+            if __file__ in line and "_glom" in line:
```

Comment:
When is `__file__` in line but `_glom` is not?

Reply:
Also does this handle going in and out of glom? (e.g., callables which call glom themselves? also customized type handlers via register_op)

Reply:
so, the goal here is "everything between the glom.glom() call at the top, and the last glom._glom() call at the bottom is replaced with glom stack"

in general I was not trying to handle cases where there is a nested call out to glom() again

I think that would probably be fine though, since the top-level glom() will intercept and re-raise -- from that point on glom._glom() is no longer in the traceback so it won't clip the stack trace too early

not sure how customized type handlers would enter the picture?

I think a dotted-line between glom and not-glom would work fine from a stack-trace coalescing perspective